Vagrantfile

```diff
@@ -0,0 +1,53 @@
+# The ip all nodes run on (nodes_ip.0/1 gives problems so start from .2)
+nodes_ip = "192.168.58."
+start_device = 2
+
+# The control node configuration
+controller_cores = 2
+controller_mem = 1024 * 4
+
+# The worker nodes configuration
+workers = 2
+worker_cores = 2
+worker_mem = 1024 * 4
+
+# Configure the VM
+Vagrant.configure("2") do |config|
+    # The box and version used
+    config.vm.box = "bento/ubuntu-24.04"
+    config.vm.box_version = "202404.26.0"
+
+    # Define the controller node
+    config.vm.define "controller" do |controller|
+        controller.vm.hostname = "controller"
+        controller.vm.network "private_network", ip: nodes_ip + "#{start_device}"
```

Comment:
The ip part of this could be `"#{nodes_ip}#{start_device}"`

Reply:
Improved this